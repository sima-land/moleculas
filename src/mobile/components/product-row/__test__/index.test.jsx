import React from 'react';
import { mount } from 'enzyme';
import { ProductRow } from '..';
import { noop } from 'lodash';
import { fireEvent } from '@testing-library/react';

describe('ProductRow', () => {
  const item = {
    itemUrl: 'test',
    imageUrl: 'https://cdn3.static1-sima-land.com/items/29455/1/140.jpg?v=1598850721',
    name: 'Зеленый слоник 25мм МИКС',
    sid: 123,
    count: 23,
    unit: 'шт.',
    initialCount: 42,
    currencyGrapheme: '₽',
    commonPrice: 50,
    itemPrice: 5,
    onWishButtonClick: noop,
    onActionsClick: noop,
    isFetchingWishItems: noop,
    notInStock: true,
    isWished: true,
    movedOrderId: 123456,
    deliveryCost: 10,
    flags: [],
  };

  it('renders properly', () => {
    const wrapper = mount(<ProductRow {...item} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render wish button', () => {
    const spy = jest.fn();
    const wrapper = mount(<ProductRow {...item} onWishButtonClick={spy} />);
    expect(wrapper.getDOMNode().querySelectorAll('.wish-button')).toHaveLength(1);
  });

  it('should render wish button with click handler', () => {
    const spy = jest.fn();
    const wrapper = mount(
      <ProductRow {...item} isFetchingWishItems={false} onWishButtonClick={spy} isWished />,
    );

    expect(wrapper.getDOMNode().querySelectorAll('.wish-button')).toHaveLength(1);

    expect(spy).toHaveBeenCalledTimes(0);
    fireEvent.click(wrapper.getDOMNode().querySelector('.wish-button'));
    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should render wish button not pressed', () => {
    const spy = jest.fn();
    const wrapper = mount(
      <ProductRow {...item} isFetchingWishItems={false} onWishButtonClick={spy} isWished={false} />,
    );

    expect(wrapper.getDOMNode().querySelectorAll('.wish-button')).toHaveLength(1);

    expect(spy).toHaveBeenCalledTimes(0);
    fireEvent.click(wrapper.getDOMNode().querySelector('.wish-button'));
    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should NOT render wish button', () => {
    const wrapper = mount(<ProductRow {...item} onWishButtonClick={undefined} />);
    expect(wrapper.getDOMNode().querySelectorAll('.wish-button')).toHaveLength(0);
  });
});
