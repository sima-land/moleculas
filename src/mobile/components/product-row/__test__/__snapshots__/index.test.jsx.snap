// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow renders properly 1`] = `
<ProductRow
  commonPrice={50}
  count={23}
  currencyGrapheme="₽"
  deliveryCost={10}
  flags={Array []}
  imageUrl="https://cdn3.static1-sima-land.com/items/29455/1/140.jpg?v=1598850721"
  initialCount={42}
  isFetchingWishItems={[Function]}
  isWished={true}
  itemPrice={5}
  itemUrl="test"
  movedOrderId={123456}
  name="Зеленый слоник 25мм МИКС"
  notInStock={true}
  onActionsClick={[Function]}
  onWishButtonClick={[Function]}
  sid={123}
  unit="шт."
>
  <div
    className="wrapper"
  >
    <div
      className="image-wrapper"
    >
      <ForwardRef(Link)
        href="test"
      >
        <a
          className="link brand-blue"
          data-testid="anchor"
          href="test"
        >
          <img
            className="image"
            src="https://cdn3.static1-sima-land.com/items/29455/1/140.jpg?v=1598850721"
          />
        </a>
      </ForwardRef(Link)>
      <ForwardRef
        checked={true}
        className="wish-button is-wished"
        disabled={[Function]}
        height={20}
        onClick={[Function]}
        width={20}
      >
        <ForwardRef(SvgFilledHeart)
          aria-label="Убрать из избранного"
          className="button wish-button is-wished checked disabled"
          data-testid="favorite-button"
          height={24}
          role="button"
          width={24}
        >
          <svg
            aria-label="Убрать из избранного"
            className="button wish-button is-wished checked disabled"
            data-testid="favorite-button"
            height={24}
            role="button"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.657 4.01C17.018 2.174 13.76 3.031 12 5.132c-1.759-2.1-5.018-2.968-7.657-1.122-1.4.98-2.279 2.632-2.339 4.376-.14 3.958 3.299 7.13 8.547 11.996l.1.091c.76.704 1.93.704 2.689-.01l.11-.102c5.248-4.855 8.676-8.027 8.546-11.985-.06-1.734-.94-3.386-2.339-4.366z"
            />
          </svg>
        </ForwardRef(SvgFilledHeart)>
      </ForwardRef>
    </div>
    <div
      className="info-column"
    >
      <ForwardRef(Link)
        className="link"
        color="gray87"
        href="test"
      >
        <a
          className="link link gray87"
          data-testid="anchor"
          href="test"
        >
          <ForwardRef(Text)
            weight={600}
          >
            <span
              className="weight-600"
            >
              Зеленый слоник 25мм МИКС
            </span>
          </ForwardRef(Text)>
        </a>
      </ForwardRef(Link)>
      <Box
        marginTop={2}
      >
        <div
          className="M-t__2"
        >
          <ForwardRef(Text)
            color="gray38"
          >
            <span
              className="color__gray38"
            >
              Арт.: 
              123
            </span>
          </ForwardRef(Text)>
        </div>
      </Box>
      <Box
        marginTop={2}
      >
        <div
          className="M-t__2"
        >
          <ForwardRef(Text)
            element="div"
            size={12}
          >
            <div
              className="size-12 line-height-16"
            >
              <div>
                Было в заказе: 42 шт.
              </div>
              <div>
                Вы получите: 23 шт.
              </div>
            </div>
          </ForwardRef(Text)>
        </div>
      </Box>
      <Box
        marginTop={2}
      >
        <div
          className="M-t__2"
        >
          <ForwardRef(Text)
            element="div"
            size={12}
            weight={600}
          >
            <div
              className="size-12 line-height-16 weight-600"
            >
              <Price
                currencyGrapheme="₽"
                value={5}
              >
                <span
                  className="root"
                  data-testid="price"
                >
                  5 ₽
                </span>
              </Price>
               × 23 шт.
            </div>
          </ForwardRef(Text)>
        </div>
      </Box>
      <Box
        marginTop={4}
      >
        <div
          className="M-t__4"
        >
          <ForwardRef(Text)
            element="div"
            weight={600}
          >
            <div
              className="weight-600"
            >
              <Price
                currencyGrapheme="₽"
                value={50}
              >
                <span
                  className="root"
                  data-testid="price"
                >
                  50 ₽
                </span>
              </Price>
            </div>
          </ForwardRef(Text)>
        </div>
      </Box>
      <Box
        marginTop={2}
      >
        <div
          className="M-t__2"
        >
          <ForwardRef(Text)
            color="gray38"
            element="div"
          >
            <div
              className="color__gray38"
            >
              Перемещен в заказ
            </div>
          </ForwardRef(Text)>
          <ForwardRef(Link)>
            <a
              className="link brand-blue"
              data-testid="anchor"
            >
              №
              123456
            </a>
          </ForwardRef(Link)>
        </div>
      </Box>
      <Box
        marginTop={2}
      >
        <div
          className="M-t__2"
        >
          <ForwardRef(Text)
            color="gray38"
            element="div"
          >
            <div
              className="color__gray38"
            >
              Нет на складе
            </div>
          </ForwardRef(Text)>
        </div>
      </Box>
    </div>
    <div
      className="action-button"
    >
      <ForwardRef(MoreSVG)
        onClick={[Function]}
      >
        <svg
          height={24}
          onClick={[Function]}
          viewBox="0 0 24 24"
          width={24}
        >
          <path
            clipRule="evenodd"
            d="M16 12c0 1.1.9 2 2 2s2-.9 2-2-.9-2-2-2-2 .9-2 2zM14 12c0-1.1-.9-2-2-2s-2 .9-2 2 .9 2 2 2 2-.9 2-2zM8 12c0-1.1-.9-2-2-2s-2 .9-2 2 .9 2 2 2 2-.9 2-2z"
            fillRule="evenodd"
          />
        </svg>
      </ForwardRef(MoreSVG)>
    </div>
  </div>
</ProductRow>
`;

exports[`ProductRow renders properly without heart 1`] = `
<ProductRow
  commonPrice={50}
  count={23}
  currencyGrapheme="₽"
  deliveryCost={10}
  flags={Array []}
  imageUrl="https://cdn3.static1-sima-land.com/items/29455/1/140.jpg?v=1598850721"
  initialCount={42}
  isFetchingWishItems={[Function]}
  isWished={true}
  itemPrice={5}
  itemUrl="test"
  movedOrderId={123456}
  name="Зеленый слоник 25мм МИКС"
  notInStock={true}
  onActionsClick={[Function]}
  onWishButtonClick={null}
  sid={123}
  unit="шт."
>
  <div
    className="wrapper"
  >
    <div
      className="image-wrapper"
    >
      <ForwardRef(Link)
        href="test"
      >
        <a
          className="link brand-blue"
          data-testid="anchor"
          href="test"
        >
          <img
            className="image"
            src="https://cdn3.static1-sima-land.com/items/29455/1/140.jpg?v=1598850721"
          />
        </a>
      </ForwardRef(Link)>
    </div>
    <div
      className="info-column"
    >
      <ForwardRef(Link)
        className="link"
        color="gray87"
        href="test"
      >
        <a
          className="link link gray87"
          data-testid="anchor"
          href="test"
        >
          <ForwardRef(Text)
            weight={600}
          >
            <span
              className="weight-600"
            >
              Зеленый слоник 25мм МИКС
            </span>
          </ForwardRef(Text)>
        </a>
      </ForwardRef(Link)>
      <Box
        marginTop={2}
      >
        <div
          className="M-t__2"
        >
          <ForwardRef(Text)
            color="gray38"
          >
            <span
              className="color__gray38"
            >
              Арт.: 
              123
            </span>
          </ForwardRef(Text)>
        </div>
      </Box>
      <Box
        marginTop={2}
      >
        <div
          className="M-t__2"
        >
          <ForwardRef(Text)
            element="div"
            size={12}
          >
            <div
              className="size-12 line-height-16"
            >
              <div>
                Было в заказе: 42 шт.
              </div>
              <div>
                Вы получите: 23 шт.
              </div>
            </div>
          </ForwardRef(Text)>
        </div>
      </Box>
      <Box
        marginTop={2}
      >
        <div
          className="M-t__2"
        >
          <ForwardRef(Text)
            element="div"
            size={12}
            weight={600}
          >
            <div
              className="size-12 line-height-16 weight-600"
            >
              <Price
                currencyGrapheme="₽"
                value={5}
              >
                <span
                  className="root"
                  data-testid="price"
                >
                  5 ₽
                </span>
              </Price>
               × 23 шт.
            </div>
          </ForwardRef(Text)>
        </div>
      </Box>
      <Box
        marginTop={4}
      >
        <div
          className="M-t__4"
        >
          <ForwardRef(Text)
            element="div"
            weight={600}
          >
            <div
              className="weight-600"
            >
              <Price
                currencyGrapheme="₽"
                value={50}
              >
                <span
                  className="root"
                  data-testid="price"
                >
                  50 ₽
                </span>
              </Price>
            </div>
          </ForwardRef(Text)>
        </div>
      </Box>
      <Box
        marginTop={2}
      >
        <div
          className="M-t__2"
        >
          <ForwardRef(Text)
            color="gray38"
            element="div"
          >
            <div
              className="color__gray38"
            >
              Перемещен в заказ
            </div>
          </ForwardRef(Text)>
          <ForwardRef(Link)>
            <a
              className="link brand-blue"
              data-testid="anchor"
            >
              №
              123456
            </a>
          </ForwardRef(Link)>
        </div>
      </Box>
      <Box
        marginTop={2}
      >
        <div
          className="M-t__2"
        >
          <ForwardRef(Text)
            color="gray38"
            element="div"
          >
            <div
              className="color__gray38"
            >
              Нет на складе
            </div>
          </ForwardRef(Text)>
        </div>
      </Box>
    </div>
    <div
      className="action-button"
    >
      <ForwardRef(MoreSVG)
        onClick={[Function]}
      >
        <svg
          height={24}
          onClick={[Function]}
          viewBox="0 0 24 24"
          width={24}
        >
          <path
            clipRule="evenodd"
            d="M16 12c0 1.1.9 2 2 2s2-.9 2-2-.9-2-2-2-2 .9-2 2zM14 12c0-1.1-.9-2-2-2s-2 .9-2 2 .9 2 2 2 2-.9 2-2zM8 12c0-1.1-.9-2-2-2s-2 .9-2 2 .9 2 2 2 2-.9 2-2z"
            fillRule="evenodd"
          />
        </svg>
      </ForwardRef(MoreSVG)>
    </div>
  </div>
</ProductRow>
`;
